R/gh_cli.R

```diff
@@ -0,0 +1,50 @@
+# See cli manual at https://cli.github.com/manual/
+
+# This functions tries the gh command in a terminal. If it errors, the gh cli
+# isn't available on the machine or at least not on the PATH variable
+gh_cli_available <- function(){
+  gh_test <- try(system2("gh", stderr = TRUE, stdout = FALSE), silent = TRUE)
+
+  if ( inherits(gh_test, "try-error") ){
+    cli::cli_alert_warning("The gh cli is not available on your machine!")
+    return(FALSE)
+  } else {
+    return(TRUE)
+  }
+}
+
+gh_cli_release_upload <- function(files,
+                                  tag,
+                                  ...,
+                                  repo = "nflverse/nflverse-data",
+                                  overwrite = TRUE){
+  # see https://cli.github.com/manual/gh_release_upload
+
+  # validate file paths
+  file_available <- file.exists(files)
+  if ( !all(file_available) ){
+    cli::cli_abort("The following file{?s} {?is/are} missing: {.path {files[!file_available]}}")
```

Comment:
I am not perfectly happy with this behavior. Maybe we should warn about missing files and go on with the code IF there are any valid file names

Reply:
DONE